src/main/java/io/harness/cf/client/api/StorageRepository.java

```diff
@@ -0,0 +1,208 @@
+package io.harness.cf.client.api;
+
+import io.harness.cf.client.common.Cache;
+import io.harness.cf.client.common.Storage;
+import io.harness.cf.model.Clause;
+import io.harness.cf.model.FeatureConfig;
+import io.harness.cf.model.Segment;
+import io.harness.cf.model.ServingRule;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+
+@Slf4j
+class StorageRepository implements Repository {
+  private final Cache cache;
+  private Storage store;
+  private final RepositoryCallback callback;
+
+  public StorageRepository(@NonNull Cache cache, RepositoryCallback callback) {
```

Comment:
Is there a reason for repo without Storage? if there is I would still make the storage final (make class immutable), but give it null value (or maybe make it Optional - though Optional use is would be debatable in this use case)

Reply:
yes, we have customers who only use cache